File: app/src/main/java/com/tangem/data/network/AdaliteApi.java
package com.tangem.data.network;

import com.tangem.data.network.model.AdaliteBody;
import com.tangem.data.network.model.AdaliteResponse;
import com.tangem.data.network.model.AdaliteResponseUtxo;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.Headers;
import retrofit2.http.POST;
import retrofit2.http.Path;

public interface AdaliteApi {
    @GET(ServerApiAdalite.ADALITE_ADDRESS)
    Call<AdaliteResponse> adaliteAddress(@Path("address") String address);

    @Headers("Content-Type: application/json")
    @POST(ServerApiAdalite.ADALITE_UNSPENT_OUTPUTS)
    Call<AdaliteResponseUtxo> adaliteUnspent(@Body String address);

//    @GET(ServerApiAdalite.ADALITE_TRANSACTION)
//    Call<AdaliteResponse> adaliteTransaction(@Path("txId") String txId);

    @Headers("Content-Type: application/json")
    @POST(ServerApiAdalite.ADALITE_SEND)
    Call<String> adaliteSend(@Body AdaliteBody adaliteBody);
}

File: blockchain/src/main/java/com/tangem/blockchain/blockchains/cardano/network/adalite/AdaliteApi.kt
package com.tangem.blockchain.blockchains.cardano.network.api

import com.tangem.blockchain.blockchains.cardano.network.adalite.AdaliteAddress
import com.tangem.blockchain.blockchains.cardano.network.adalite.AdaliteSendBody
import com.tangem.blockchain.blockchains.cardano.network.adalite.AdaliteUnspents
import retrofit2.http.*

interface AdaliteApi {
    @GET("/api/addresses/summary/{address}")
    suspend fun getAddress(@Path("address") address: String): AdaliteAddress

    @Headers("Content-Type: application/json")
    @POST("/api/bulk/addresses/utxo")
    suspend fun getUnspents(@Body addresses: List<String>): AdaliteUnspents

    @Headers("Content-Type: application/json")
    @POST("/api/v2/txs/signed")
    suspend fun sendTransaction(@Body adaliteBody: AdaliteSendBody): String // List<Any>?
}